Tabs usage a11y updates (#4070)
carbon-design-system/carbon-website: src/pages/components/tabs/usage.mdx

### When not to use

#### Filtering the same content

> When toggling between between different formats of the same content or filtering

Reviewer: ```suggestion
When toggling between different formats of the same content or filtering
```

#### 1. Line tabs

> A. Label <br /> B. Indicator <br /> C. Scroll button

Author: I haven't updated this quite yet—i was starting to label the indicator selected indicator and i went back to just indicator because i wondering if we would need to call out the unselected indicator and crowd the image. i could update if it's best to call out both.

For further content guidance, see Carbon's
[content guidelines](/guidelines/content/overview).

## Universal behaviors

### States

Tabs have two states—**selected,** and **unselected,**. Other interactive states
are **hover, focus,** and **disabled**. The default view is one tab is
preselected and is usually the first tab. Only one tab can be selected at a
time. When a user chooses a new item, the previous tab is automatically

> deselected. If a user navigates away from a tab, a user should return to the

Reviewer: >  If a user navigates away from a tab, a user should return to the last tab selected.

Michael suggested changing this sentence to say this "If a user navigates away from a tab, it remains the selected tab until altered by the user." You can make the call if you want to change it or not to something like this.

last tab selected.

For detailed visual information about the various states for this component, see
the [Style tab](https://carbondesignsystem.com/components/tabs/style).

| State      | When to use                                                                                                                                                                                                                                              |
| ---------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

> | Selected   | When a user clicks or uses the arrows keys to activate tab.                                                                                                                                                                                              |

Reviewer: ```suggestion
| Selected   | When a user clicks or uses the arrows keys to activate the tab.                                                                                                                                                                                              |
```

> | Hover      | When a user is hovering over the tag with the mouse cursor to interact with it.                                                                                                                                                                          |

Reviewer: ```suggestion
| Hover      | When a user is hovering over the tab with the mouse cursor to interact with it.                                                                                                                                                                          |
```

> | Focus      | When a user clicks on the tag or navigates using the keyboard with left and right arrows, it becomes focused, indicating the user has successfully navigated to the component.                                                                           |

Reviewer: ```suggestion
| Focus      | When a user clicks on the tab or navigates using the keyboard with left and right arrows, it becomes focused, indicating the user has successfully navigated to the component.                                                                           |
```

| Unselected | When the user navigates to another tab and the tab is inactive.                                                                                                                                                                                          |
| Disabled   | When a user is not allowed to interact with the tab due to either permissions, dependencies, or pre-requisites. The disabled state completely removes the interactive function from a component. The styling is not subject to WCAG contrast compliance. |

<Row>
<Column colLg={8}>

> ![Example of line and contained tab selected, unselected, and disabled states](images/tab-usage-10.png)

Reviewer: I think we could remove this image if you want. It also doesn't currently show the hover and focus states that are listed in the table above. We have started to do this on some of the Usage tabs when we provide a link out to the Style tab for people to reference the more in-depth state images. I think we are starting to especially do this when there are many variants of one component.

pressing right or left arrow keys. For additional keyboard interactions, see the
[Accessibility tab](https://carbondesignsystem.com/components/tabs/accessibility).

#### Automatic and manual

Automatic and manual tablists differ in how the tab items are activated.

For automatic tablists, focus and selection are synchronized. When the user

> arrows to a tab, it becomes selected, and the tab panel under the tab is updates

Reviewer: ```suggestion
arrows to a tab, it becomes selected, and the tab panel under the tab updates
```

automatically.

Manual tablists allow the user to arrow between the tab items without updating
the tab panel underneath. When the user navigates using arrows, the selected tab
remains selected while focus moves to the next tab. In order to select the

> focused tab (and update the tabpanel under the tab), the user would press Enter

Reviewer: ```suggestion
focused tab (and update the tabpanel under the tab), the user would press `Enter`
```

> or Space.

Reviewer: ```suggestion
or `Space`.
```

<Row>
<Column colLg={8}>

<Tabs>

<Tab label="Automatic">

> <GifPlayer color='dark'>

Reviewer: Regarding the gifs: 

— Does the magenta annotation apply to an individual tab at a time? I am wondering if the annotation needs to sit above the tab that is changes or being interacted with in each frame. 

— I also think the gifs should be standalone and go under each individual content section on the page. I think you are right it might be a lot of interaction in the Gatsby tab image component. 

— If you are still experiencing a white border bug with using the Gatsby gif component try reaching out to Thy, she has been working with it recently. However, I think that may be happening because of how the Gatsby tab image component is structured.

![Example of the tab behavior of automatic tabs](/images/tab-usage-12a-animated.gif)

![Example of the tab behavior of automatic tabs](/images/tab-usage-12a.png)

</GifPlayer>

</Tab>

<Tab label="Manual">

<GifPlayer color='dark'>

![Example of the tab behavior of manual tabs](/images/tab-usage-12b-animated.gif)

![Example of the tab behavior of manual tabs](/images/tab-usage-12b.png)

</GifPlayer>

</Tab>

</Tabs>

</Column>
</Row>

<Caption>

Automatic and manual tab interactions.

</Caption>

For additional information, see the
[Accessibility tab](https://carbondesignsystem.com/components/tabs/accessibility).

## Line tabs

A line tab is standalone tab that can also be nested within components. Although
it is commonly used within components, such as a modal or header, line tabs can
also be used for large content areas using the entire page for layout, not
connected to any other components. Line tabs are highly flexible and can be
placed on background or layer tokens, making it easy to use in many different

> scenarios. To help define the tabpanel, which is the content area below the

Reviewer: ```suggestion
scenarios. To help define the tab panel, which is the content area below the
```

selected tab, a line can be used that extends from the tablist to the end of the
panel. This is currently not part of the Carbon line tabs component and must be

> added by teams is desired.

Reviewer: ```suggestion
added by teams if desired.
```

<Row>
<Column colLg={12}>

![Examples of tabs with icons](images/tab-usage-13.png)

</Column>
</Row>

## Contained tabs

Contained tabs are an emphasized tab commonly used for defined content areas

> like sub-pages. Contained tabs are always attached to a tabpanel (background

Reviewer: ```suggestion
like sub-pages. Contained tabs are always attached to a tab panel (background
```

container) that uses the same layer token as the selected tab. Since the

> contained tabs are on layers, the tab content tends stands out against the

Reviewer: ```suggestion
contained tabs are on layers, the tab content tends to stand out against the
```

> background and maintains a high visual hierarchy while line tabs tend to blend

Reviewer: ```suggestion
background and maintain a high visual hierarchy while line tabs tend to blend
```

<Row>
<Column colLg={8}>

![Examples of contained tabs with icons](images/tab-usage-15.png)

</Column>
</Row>

<Caption>Example of contained tabs with icons.</Caption>

> ### Icon-only tabs

Reviewer: In this "Icon-only tabs" section, it just needs to add the word "focus" to this sentence:

"Always use a tooltip for an icon description on hover and focus to add clarity."
